avocado/utils/process.py

```diff
@@ -219,7 +219,7 @@ def get_children_pids(ppid, recursive=False):
     return children
 
 
-def binary_from_shell_cmd(cmd):
+def binary_from_shell_cmd(cmd, encoding=None):
```

Comment:
This is in a big part, a revert of 1401f3cfd959402a24ab794317a752549d957de2.  Before considering reverting the general direction of Unicode usage, and the of the Avocado's APIs, I have to zero down on one or more examples described in your commit message.  I need to understand if the issue you're reporting has to do with the command not being unicode, or the command output.

Reply:
Basically you can test this easily by forcing different encoding. Let's say you execute on a remote machine, or the application uses a different encoding. It's been a while since I wrote that but to test it I used multiple layers of shells with different encoding (ISO, cp1250, UTF8). The original code spoils the command while this version makes sure the proper command is executed (I debugged the issues to subprocess, when string is passed it's encoded there but in a different encoding that the one we are suppose to use).

Reply:
Also see the new selftests that does the different-encoding checks.

Reply:
@ldoktor It's definitely a must to have reproducers for the issues we're trying to fix, so I'll take a look at the new selftests, and *if* they fall short of explaning/testing the issues, I'll try to improve the situation.